Chatspace front

teru0225 opened the pull request:
WHY
chatspaceを作る上で、メッセージ画面のviewが必須だから
WHAT
メッセージ画面のviewを作成しました。

<img width="1440" alt="chatspace" src="https://user-images.githubusercontent.com/45619905/51235289-776cc180-19b2-11e9-9137-dfed91f08871.png">

teru0225 committed:
Edit .gitignore

edited .gitignore

teru0225 committed:
Edit README.md

Edit README.md
- you can see the construction of the tables for Chat-space

teru0225 committed:
Edit README

There were 2 mistakes
- Unnecessary dot
- wrong space

teru0225 committed:
Edit README

deleted
- password column
(Unnecessary to write down columns that devise gem makes)
- accepted_nested_attributes_for
(Unnecessary to write down on README)

teru0225 committed:
Edit READEME

Deleted  group_id column in user table
(Unnecessary)

teru0225 committed:
Install gems

Installed
- haml-rails
- erb2haml

teru0225 committed:
Committed command "rake haml:replace_erbs"

Committed command "rake haml:replace_erbs"

The result
Converting: app/views/layouts/application.html.erb... Done!
Removing: app/views/layouts/application.html.erb... Removed!
Converting: app/views/layouts/mailer.html.erb... Done!
Removing: app/views/layouts/mailer.html.erb... Removed!
Converting: app/views/layouts/mailer.text.erb... Done!
Removing: app/views/layouts/mailer.text.erb... Removed!

teru0225 committed:
Deleted codes on application.css

In order to convert application.css file to application.scss

teru0225 committed:
Rename application.css to application.scss

teru0225 committed:
Edit application.scss

In order to read reset file

teru0225 committed:
Create _variable.scss file

In this file, you can code about colors

teru0225 committed:
Create _reset.scss

teru0225 committed:
Edit _reset.scss

teru0225 committed:
Edit application.scss

Application.scss can import
- _reset.scss
- _style.scss
- _variable.scss
- font-awesome

teru0225 committed:
Install new gem "font-awesome-rails"

teru0225 committed:
Installed new gem "font-awesome-rails"

teru0225 committed:
Created new file "index.html.haml"

teru0225 committed:
Commit command "rails g controller messages"

Created new controller file “messages”

teru0225 committed:
Define index in "messages_controller.rb"

defined index

teru0225 committed:
Installed "font-awesome-rails"

teru0225 committed:
Installed "font-awesome-rails"

teru0225 committed:
Edit routes.rb

Defined
- rooting for chatspace front

teru0225 committed:
Edit view file "index.html.haml"

I defined
<head>

<body>
- container
- left content
- left header
- right content
- right body
- right body2
- right body 3
- right footer

teru0225 committed:
Edit scss file "style.scss"

teru0225 committed:
Make blank line in every files

As the rule for coding, I needed to make one blank line in every files

teru0225 committed:
Command "rails s"

teru0225 committed:
Create new file "header.scss"

In this file, I coded scss for header

teru0225 committed:
Create new scss file "_leftcontent.scss"

Defined scss for left-content

teru0225 committed:
Deleted scss file "_style.scss"

I don’t need this file any more because I made partial files for these
codes

teru0225 committed:
Created new scss

teru0225 committed:
edited new file "variable.scss"

In this file, I code for colors

teru0225 committed:
Created new haml file "header"

I coded for header

teru0225 committed:
Edit index.htnl.haml

teru0225 committed:
Created new file main.html.haml

teru0225 committed:
Command rails s

40213151 commented on app/views/messages/index.html.haml:
```diff
@@ -1,36 +1,13 @@
+  .leftContent
+    .leftHeader
+      %p.leftHeader__name seo
+      %a.leftHeader__settingIcon{:href => "#"} <i class="fa fa-cog" aria-hidden="true"></i>
```
こちらのアイコンですが、Railsのヘルパーメソッドを使用して表示しましょう！

40213151 commented on app/views/messages/index.html.haml:
```diff
@@ -1,36 +1,13 @@
+    .leftHeader
+      %p.leftHeader__name seo
+      %a.leftHeader__settingIcon{:href => "#"} <i class="fa fa-cog" aria-hidden="true"></i>
+      %a.leftHeader__editIcon{:href => "#"} <i class="fa fa-pencil-square-o" aria-hidden="true"></i>
```
こちらのアイコンですが、Railsのヘルパーメソッドを使用して表示しましょう！

40213151 commented:
プルリクエストを出す際は、WHAT.WHYなど、プルリクの詳細が第三者からわかる様にしましょう！
(例: その機能を実装するとどの様なメリットを産むのか)
Viewの実装の際は、Gyazoを使用しスクリーンショットを載せていただくとよりわかりやすいです

teru0225 committed:
Edit index.html.haml

I had used a tag.
Instead of using a tag, I used link_to tag

teru0225 commented:
> プルリクエストを出す際は、WHAT.WHYなど、プルリクの詳細が第三者からわかる様にしましょう！
> (例: その機能を実装するとどの様なメリットを産むのか)
> Viewの実装の際は、Gyazoを使用しスクリーンショットを載せていただくとよりわかりやすいです

レビューありがとうございます。
スクリーンショットの件、かしこまりました。

ky3090 commented:
以前コメントしていただいた分は、修正していますでしょうか？

ky3090 commented:
また、cacheがgithub上に上がってきているので、.gitignoreに記述して上がらないようにしましょう。

teru0225 commented:
> 以前コメントしていただいた分は、修正していますでしょうか？

Railsのヘルパーメソッドを使うようにとのご指摘は、すでに修正済みです。

teru0225 commented:
> また、cacheがgithub上に上がってきているので、.gitignoreに記述して上がらないようにしましょう。

ターミナルにて、git rm -r --cachedコマンドを実行して、cacheが上がらないようにしました。

ta910 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,13 @@
+  .leftContent
+    .leftHeader
+      %p.leftHeader__name seo
+      %a.leftHeader__settingIcon{:href => "#"} <i class="fa fa-cog" aria-hidden="true"></i>
```
こちらヘルパーメソッド(link_to)を使う修正の反映をお願いいたします

ta910 commented:
WHY,WHAT,スクリーンショットをPRに記述しましょう！
このPRからしっかり変えていきましょう

ta910 commented:
キャッシュがあるとファイル数が多すぎてレビューが難しいため、キャッシュファイルがFile Changedに反映されないようにお願いしたいです。

キャッシュファイルを全て削除してからもう一度pushして、そのあとにgitignoreに追加する方法が良いかと思います」

teru0225 committed:
Deleted cache

Deleted unnecessary cache files

teru0225 committed:
Add cache file in .gitignore

teru0225 commented:
> キャッシュがあるとファイル数が多すぎてレビューが難しいため、キャッシュファイルがFile Changedに反映されないようにお願いしたいです。
> 
> キャッシュファイルを全て削除してからもう一度pushして、そのあとにgitignoreに追加する方法が良いかと思います」

指摘していただいた２点を改善しました。
よろしくお願いいたします。

40213151 commented on app/assets/stylesheets/_style.scss:
```diff
@@ -0,0 +1,207 @@
+                  }
+}
+
+.rightBody {
```
こちらの要素ですが、メッセージが多く投稿された分はスクロールで表示できる様にしましょう！
もし方法が分からなければTech::connectでメンターにお聞きください！

40213151 commented:
index.htmlにてヘルパーメソッド(link_to)を使用する変更は、反映されていますでしょうか？
今一度ご確認お願いします！

teru0225 committed:
Deleted cache file

teru0225 committed:
Edited _main.scss

Added overflow-y: scroll; in right body
Now when messages are too many to show on screen, users can scroll the
messages.

teru0225 commented:
> index.htmlにてヘルパーメソッド(link_to)を使用する変更は、反映されていますでしょうか？
> 今一度ご確認お願いします！

指摘いただいた点、修正いたしました。
もう一度、ご確認お願いいたします。
ちなみに、index.html.hamlは以下のようになっています。
<img width="735" alt="index" src="https://user-images.githubusercontent.com/45619905/51423831-0246f980-1c09-11e9-8c2a-a1b8d20d414b.png">

hiroki5123 commented on app/views/messages/_main.html.haml:
```diff
@@ -0,0 +1,22 @@
+    .rightHeader
+      .rightHeader__group.clearfix
+        .rightHeader_group_name sampleグループ
+        %a.rightHeader_group_edit{:href => "#"} Edit
```
ここlink_toで書きましょう

teru0225 replied:
ご指摘いただいた２点、修正しました。
再度、ご確認よろしくお願いいたします。

hiroki5123 commented on tmp/pids/server.pid:
```diff
@@ -0,0 +1 @@
+1468
```
このファイルもgitignoreしましょう

teru0225 committed:
Edited _main.html.haml

I edited .rightHeader_group_edit
I recoded that in order to use link_to method.

teru0225 committed:
Edited .gitignore

I added tmp/pids/* in .gitignore

hiroki5123 commented on log/development.log:
```diff
@@ -2188,3 +2188,66 @@ Processing by MessagesController#index as HTML
 Completed 200 OK in 1699ms (Views: 1677.7ms | ActiveRecord: 0.0ms)
 
 
```
このファイルもgitignoreしましょう

tmp以下は全てgitignoreしてしまって大丈夫です

teru0225 replied:
かしこまりました。
gitignoreは以下のようななっております。
また、development.logファイルとtmp/cache/assets以下のファイルは削除しました。
<img width="461" alt="index2" src="https://user-images.githubusercontent.com/45619905/51424189-e5f98b80-1c0d-11e9-83c8-d51da5cd0bb7.png">

teru0225 committed:
Edit .gitignore

Added
Tmp/*
log/development.log

yuki-uchida commented on app/assets/stylesheets/_main.scss:
```diff
@@ -0,0 +1,144 @@
+  padding-top: 26px;
+}
+
+.rightBody3 {
```
rightBody1,2,3という命名は、何か他にわかりやすいものを考えて見て欲しいです。
right__body--headerなど、BEMの形に沿って書いてあるとより良いです。全体的に、クラス名は修正できそうですね。

yuki-uchida replied:
https://qiita.com/Takuan_Oishii/items/0f0d2c5dc33a9b2d9cb1

yuki-uchida commented on app/assets/stylesheets/_main.scss:
```diff
@@ -0,0 +1,144 @@
+              }
+}
+
+.rightFooter {
```
こちらも、B(Block) E(Element) M(Modifer)に沿って言えば、
right__body--footerなどに出来ると思います

teru0225 replied:
ご指摘いただきありがとうございます。
送っていただいた記事を見て、BEMでの設計をし直しました。
ご確認、よろしくお願いいたします。

yuki-uchida reviewed (changes requested):
BEMのクラス設計だけもう少し修正できそうですね！

teru0225 committed:
Edited _main.scss and _main.html.haml

I edited those files in order to apply BEM rule

teru0225 committed:
Applied BEM rules in every view files
